portfolio/src/main/webapp/script.js

```diff
@@ -52,10 +52,20 @@ function addRandomFact() {
  * Fetches messages from server and adds it to DOM
  */
 function getComments() {
-  fetch('/data') 
+  try {
+    var index = document.getElementById('max-comments').selectedIndex;
+    var maxComments = document.getElementsByTagName('option')[index].value;
+  }
+  catch (e) {
+      var maxComments = 3;
+  }
+
+  var url = '/data?max-comments=' + maxComments.toString();
+
+  fetch(url) 
   .then(response => response.json())
   .then((messages) => {
     document.getElementById('message-container').innerText = messages.join('\n');
```

Comment:
It might be better HTML structure if you had new element per comment. For every message, you create a new element, fill its innerText with that comment, and append this element to a parent element that is designated to hold comments.
This could be done in a different PR.